### PR TITLE
feat(app-review): analytics

### PR DESCRIPTION
**Pull request by MuckT**
### Description

Adds the following analytics events for in app review:
- `app_review_impression`
- `app_review_error`

Also wraps `InAppReview.RequestInAppReview` in a try catch with logging on unexpected errors.

### Test plan

Unit tests updated.

### Related issues

- Fixes RET-15 

### Backwards compatibility

Yes

### CODE REVIEW

File: src/appReview/saga.ts [hunk 1]
```diff
@@ -18,9 +23,17 @@ export function* setAppReview() {
 
   // If the last interaction was less than a quarter year ago or null
   if (lastInteractionTimestamp <= now - ONE_DAY_IN_MILLIS * 91) {
-    const result = yield call(InAppReview.RequestInAppReview)
-    // If the in app review was shown, update the last interaction timestamp
-    if (result) yield put(setLastInteractionTimestamp(now))
+    try {
+      const result = yield call(InAppReview.RequestInAppReview)
+      // If the in app review was shown, update the last interaction timestamp
+      if (result) {
```
Comment (kathaypacific) [inline src/appReview/saga.ts:29]: this made me wonder what to do when `result` is false...but reading the docs, it seems like `result` is always true for iOS, and true for Android _after_ they finish the review flow. am i understanding that right? if so, it seems like we should remove this if block - we always want to update the timestamp and dispatch the impression event

**Review by kathaypacific — COMMENTED** [review]
Looks great! just one comment to check that we understand the return value of RequestInAppReview correctly :)

File: src/appReview/saga.ts (hunk 1 above)
Comment (MuckT) [reply src/appReview/saga.ts:29]: It's not quite clear to me from the docs if result is always true for iOS and Android or an error. In [this example](https://github.com/MinaSamir11/react-native-in-app-review#usage) it looks like both was are used, but I also see that the updating timestamp is occurring outside the if block.

Comment (MuckT) [reply src/appReview/saga.ts:29]: After looking a bit more I think we're safe to remove the it block as result looks like it's either: `true` or an error.
- [react-native-in-app-review Android](https://github.com/MinaSamir11/react-native-in-app-review/blob/master/android/src/main/java/com/ibits/react_native_in_app_review/AppReviewModule.java#L68-L72)
- [react-native-in-app-review iOS](https://github.com/MinaSamir11/react-native-in-app-review/blob/master/ios/RNInAppReviewIOS.swift#L9C2-L23)

Comment (MuckT) [reply src/appReview/saga.ts]: Fixed in 3544d91aff763ed658c51a9bffa57f7521f1424d!

File: src/analytics/Properties.tsx [hunk 2]
```diff
@@ -1211,6 +1212,13 @@ interface TokenBottomSheetEventsProperties {
   }
 }
 
+interface InAppReviewEventsProperties {
```
Comment (kathaypacific) [inline src/analytics/Properties.tsx:1215]: i think that this can be part of AppEventsProperties right?

File: src/app/saga.test.ts [hunk 3]
```diff
@@ -547,61 +548,11 @@ describe('runCentralPhoneVerificationMigration', () => {
...
-      .put(setSupportedBiometryType(BIOMETRY_TYPE.TOUCH_ID))
-      .run()
-
-    expect(initializeSentry).toHaveBeenCalledTimes(1)
-    expect(ValoraAnalytics.init).toHaveBeenCalledTimes(1)
-    expect(initI18n).toHaveBeenCalledWith('en-US', true, '1')
-  })
-})
-
+// Keep this test before the app init test otherwise the error spec will fail
```
Comment (kathaypacific) [inline src/app/saga.test.ts:551]: this is not okay 🙈 we should never have to place unit tests in a specific order. if this is a requirement, it means that there are interdependencies between tests which is a bad thing for a unit test! in this case i think that we have not cleared up the mocks from the app init tests, and that has caused the next tests to fail...

**Review by kathaypacific — COMMENTED** [review]
looking good :) seeing your comment in the test file got me curious. i think #3716 will enable you to remove the workaround

File: src/analytics/Properties.tsx (hunk 2 above)
Comment (MuckT) [reply src/analytics/Properties.tsx]: Fixed in 24e81d31a615d6f02a97c876293433aee9eae3c6!

**Pull request merged**